### PR TITLE
Extended support for ppc64le

### PR DESCRIPTION
**Pull request by ArchaB**
Hi,

Presto-main failed on ppc64le system . Failure log seen was

**2016-06-30T10:40:48.699+0545    INFO    pool-200-thread-2       stderr  Presto requires x86-64 or amd64 on Linux (found ppc64le)**

Extended the if loop to add support for ppc64le.

Regards.

### CODE REVIEW

**Comment by facebook-github-bot-0** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by electrum** [comment]
We shouldn't extend this check until everything is fully supported on this architecture (i.e. this is the last step to supporting it).

File: presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ public static void verifyJvmRequirements()
         String osName = StandardSystemProperty.OS_NAME.value();
         String osArch = StandardSystemProperty.OS_ARCH.value();
         if ("Linux".equals(osName)) {
-            if (!"amd64".equals(osArch)) {
+	    if (!"amd64".equals(osArch) && !"ppc64le".equals(osArch)) {
                 failRequirement("Presto requires x86-64 or amd64 on Linux (found %s)", osArch);
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java]: The message should be updated, too

Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java]: This could say "Presto requires x86-64 (amd64) or ppc64le ..."

**Comment by dain** [comment]
To support ppc64le, I think we need something equivalent to TravisCI that runs on power.

**Comment by electrum** [comment]
@ArchaB is there a public CI system like Travis that supports ppc64le?

**Comment by ArchaB** [comment]
We are working on an external CI setup which would be Jenkins based. Till then, we will be setting a local CI engine in our lab and can share the results if needed

**Comment by ArchaB** [comment]
CLA signed but the '**_Merge pull request_**' button is still disabled, with the note "**You're not authorized to merge this pull request.**". 
Can anyone help on how can this changed be merged. 
Thanks

**Comment by facebook-github-bot-3** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

**Comment by brian-rickman** [comment]
@ArchaB Only Facebook employees have access to the "Merge pull request" button.  You just have to be patient.

**Comment by electrum** [comment]
Please see my first comment. Merging this will be the last step. We don't want to enable it until the platform is fully supported, which includes having a public CI system that runs for every commit (or better, before every commit, like Travis CI).

**Comment by asowani** [comment]
@electrum I would like check more about the CI setup requirement w.r.t. communication on this thread. Is it okay if we create a Jenkins build job for Presto on ppc64le and send you links? Or are you willing to setup the job yourself if ppc64le infrastructure is made available over the Internet? Or if you are looking for any specific CI only?
Thanks!

**Comment by electrum** [comment]
To clarify, any CI is fine as long as it can build each Pull Request like Travis does.

**Comment by electrum** [comment]
We need something that can run pre-commit with GitHub integration, so that people know they are breaking it before commit, similar to Travis CI. Trying to chase down breakage post-commit is much harder and makes it impossible to test changes by submitting a PR.

I know there’s a GitHub plugin for Jenkins that can build each Pull Request and update the GitHub PR status. If this can be setup for Presto, we can work on supporting it. It would be best if you can set this up and manage it as we don’t have team bandwidth for that right now.

**Comment by asowani** [comment]
@electrum Sure, we can setup Jenkins based CI for now. I think this will also take care of CI requirement for https://github.com/prestodb/presto/issues/6263 (presto-hive). I would also like to know if you would add a check point like "ppc64le build passed" to the existing pre-requisite checklist for PRs.

**Comment by electrum** [comment]
Ideally, yes, this would be another requirement for merging PRs. It will need to be reliable and fast enough*, though most PRs will not be impacted as they are touching pure Java code.

We have a few dependent projects that also need this CI, as they have native code which is definitely impacted:

https://github.com/prestodb/presto-hadoop-apache2
https://github.com/airlift/testing-mysql-server
https://github.com/airlift/testing-postgresql-server

* (It won’t be as fast as Travis which has been carefully tuned to run parts of the build/tests in parallel, but that’s fine and expected)

**Comment by electrum** [comment]
@ghatwala Thanks, closed

**Pull request closed without merging**